### ISSUE TITLE
Two values computed and stored but not read at the point of use (`centralized_ppo.update`)

### ISSUE THREAD

**Issue by AdityaKapoor74** [scout-comm/scout#1]
Hello — thank you for releasing this code. We are re-implementing SCoUT as a baseline and
have been checking our port against yours module by module. Two things in
`src/algos/scout/centralized_ppo.py` look like they may not be doing what was intended,
and we would rather ask than assume.

**1. The recipient mask at update time**

At act time the recipient distribution is masked by `recv_mask` (no self, plus the
environment's constraints), and `RolloutBuffer` stores that mask per step.

In `update()`, the mask appears to be reconstructed rather than read:

```python
recv_mask_f = G_f > 0
```

Since `G = Y Yᵀ` is a soft affinity with strictly positive entries, `G_f > 0` is
all-`True`, so the recipient head's importance ratio is evaluated over the full agent set
— including self — while the behaviour policy sampled from the masked set.
`batch["recv_mask"]` does not appear to be read anywhere in `update()`.

Is the reconstruction intentional (e.g. to keep the soft log-mask the only constraint), or
was `batch["recv_mask"]` meant to be used here?

**2. The GAE bootstrap in `update()`**

`update()` falls back to computing GAE itself when `returns`/`A_env` are absent from the
batch:

```python
last_value = torch.zeros(A, device=device)
if maybe_returns is None or maybe_A_env is None:
    returns, adv_env = compute_gae(rewards, values_old, dones, gamma, lam, last_value)
```

As far as we can tell that branch always runs — we could not find any assignment to
`batch["returns"]` or `batch["A_env"]` anywhere in the repository. Meanwhile the
bootstrap value *is* computed and stored: `scout_trainer.py` calls
`buf.set_last_value(last_v)`, and `buffers.py` places it in the batch as
`batch["last_value"]`. It appears to be read in only one place — the
`ablation_no_counterfactual` branch — which feeds the communication heads rather than the
environment advantage.

The effect would be that the environment advantage treats every iteration boundary as
terminal. Is the zero intentional (e.g. because episodes are expected to align with the
horizon), or should `batch["last_value"]` be passed here?

Happy to send a patch for either if useful. We are replicating both behaviours in our
port by default, since your published results were produced with the code as it stands,
and we would not want our baseline to be a different algorithm from the one your numbers
describe.
